integration/update_test.go

```diff
@@ -72,3 +73,194 @@ func TestUpdateUpsert(t *testing.T) {
+			result: bson.D{{"_id", "double-nan"}, {"value", math.NaN()}},
+		},
+		"DoubleNegativeIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(-42.13)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(0)}},
+		},
+		"DoubleIncrementIntField": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(1.13)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", float64(43.13)}},
+		},
+		"DoubleIncrementLongField": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", float64(1.13)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", float64(43.13)}},
+		},
+		"DoubleIntIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"DoubleLongIncrement": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"IntIncrement": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(43)}},
+		},
+		"IntNegativeIncrement": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(-1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(41)}},
+		},
+		"IntIncrementDoubleField": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"IntIncrementLongField": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int32(1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(43)}},
+		},
+		"LongIncrement": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(43)}},
+		},
+		"LongNegativeIncrement": {
+			filter: bson.D{{"_id", "int64"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(-1)}}}},
+			result: bson.D{{"_id", "int64"}, {"value", int64(41)}},
+		},
+		"LongIncrementDoubleField": {
+			filter: bson.D{{"_id", "double"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "double"}, {"value", float64(43.13)}},
+		},
+		"LongIncrementIntField": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"value", int64(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int64(43)}},
+		},
+
+		"FieldNotExist": {
+			filter: bson.D{{"_id", "int32"}},
+			update: bson.D{{"$inc", bson.D{{"foo", int32(1)}}}},
+			result: bson.D{{"_id", "int32"}, {"value", int32(42)}, {"foo", int32(1)}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t, shareddata.Scalars)
```

Comment:
I would suggest the same here - double, double-nan, int34, and int64 are much less than shareddata.Scalars.

Reply:
Same as above.

Reply:
I'm sorry I just saw in my own PR, where a lot of data was not necessary and it was taken too long for tests to retest.
And, my second argument, is that also we'd probably get rid someday of shareddata, it's in a comment of shared-data.

Reply:
Updated.